README.md

```diff
@@ -199,28 +230,39 @@ The package.json in the root of the repo has all of the required dev dependancie
 npm i <package-name> --saveDev
 ```
 
-### Updating composer dependencies
+### The S3 monitor
+
+The cypress test suite runs an instance of the S3 monitor. This is a Python application which polls an S3 bucket looking for emails sent during test runs.
+
+All emails used during test runs, such as the addresses used for new user accounts, are in the lpa.opg.service.justice.gov.uk domain. This enables us to do the following:
+
+1. During a test run, we send requests to the real Notify service, which then sends out emails on our behalf. Examples are account activation and password reset emails. These emails pass through standard email infrastructure; however, the email server for the lpa.opg.service.justice.gov.uk domain is rigged so that any emails sent to it end up in an S3 bucket (opg-lpa-casper-mailbox). (NB "casper" crops up for historical reasons, as the tests were previously implemented in casper.)
+2. The S3 monitor polls the S3 bucket, copying any messages it finds into a local directory. The filename includes the ID of the test account.
+3. Automated tests poll the local directory, looking for text files with specific IDs in their names (put there by the S3 monitor). The ID is included as part of the email address used to sign up or login a user, such as caspertests+1680708521545893617@lpa.opg.service.justice.gov.uk.
+4. On finding the correct file (once it's been pulled down from S3), the email it contains is parsed for the appropriate links. For example, while testing sign up, the cypress tests look for a link to activate the newly-created account; once found, the link is followed to mimic a user opening the link from their email client, activating the account.
+
+## Updating composer dependencies
 
-Composer install is run when the app containers are built, and on a standard `make dc-up`.
+Composer installs PHP dependencies when the app containers are built, and on a standard `make dc-up`.
 
 However, if you upgrade a package in composer.json for one or more services, you'll need to update the corresponding lock file(s).
 
 This can be done with:
 
 ```bash
-docker run -v `pwd`/service-front/:/app/ composer update --prefer-dist --no-interaction --no-scripts --ignore-platform-reqs
+docker run -v `pwd`/service-front/:/app/ composer update --prefer-dist --no-interaction --no-scripts
 ```
 
 (replacing `service-front` with the path to the application component you are adding a package to; note that you'll need to do this for the following commands as well)
 
 Packages can be added with:
 
 ```bash
-docker run -v `pwd`/service-front/:/app/ composer require author/package --prefer-dist --no-interaction --no-scripts --ignore-platform-reqs
```

Comment:
There is also in Makefile targets of run-front-composer  , run-api-composer etc.  At the mo' these ignore platform reqs. So maybe Makefile needs updating too?  Is it worth taking the docker command out of README and directing people to simply.   make run-front-composer for example  ,  in order to not have the docker command in 2 places which might drift out with each other?

Reply:
This is specifically for adding a new package, which we don't do very often. I will remove any `--ignore-platform-reqs` flags.

Reply:
👍

Reply:
The only problem with this is that there are no composer images which allow you to specify the PHP version. This means you could potentially be updating composer deps with the wrong PHP version.

Reply:
Another issue is that removing `--ignore-platform-reqs` means that you have to have any required extensions installed in the image, too. The composer images don't have the intl extension, for example, which means that you can't update the package versions if `--ignore-platform-reqs` is removed.

Reply:
I am going to suggest this is left for a separate ticket. For now, I'll reinstate `--ignore-platform-reqs`.

Reply:
I've figured it out. We use a multi-stage build, and locally use the first part of the build, leaving composer installed. When in CI, we let the whole Dockerfile run, which removes the composer files. It's pretty straightforward and non-intrusive, so I will have a go at implementing this approach this afternoon.

Reply:
Raised LPAL-1196 with a suggestion to implement the dev image approach.

Reply:
👍